### PR TITLE
feat(instant-sync): adding in instant sync

### PR DESCRIPTION
**Pull request by bassrock**
## Summary
* implement instant sync into the Pocket iOS Client

## References 
* https://getpocket.atlassian.net/browse/IN-278

## Implementation Details
* This required removing the publisher on the Session object and moving to NotificationCenter to determine login and log out.
*  Also create a few protocols for handling notifcations

## Test Steps
* See https://getpocket.atlassian.net/browse/IN-278

## PR Checklist:
- [x] Added Unit / UI tests
- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

### CODE REVIEW

File: PocketKit/Tests/SharedPocketKitTests/KeychainStorageTests.swift [hunk 1]
```diff
@@ -107,24 +107,4 @@ class KeychainStorageTests: XCTestCase {
         _ = storage.wrappedValue
         XCTAssertEqual(keychain.copyMatchingCalls.count, 3)
     }
-
-    func test_projectedValue_removesDuplicates() {
```
Comment (bassrock) [inline PocketKit/Tests/SharedPocketKitTests/KeychainStorageTests.swift:111]: This test was failing by removing the publisher on the session

**Comment by bassrock** [comment]
@jacobsimeon feel free to review and merge this and the feature branch while I am out!

File: PocketKit/Sources/PocketKit/Notifications/PocketBraze.swift [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        center.requestAuthorization(options: [.badge, .sound, .alert, .provisional]) { granted, error in
+            print("Notification authorization, granted: \(granted), error: \(String(describing: error))")
+        }
+    }
+
+    func loggedOut(session: SharedPocketKit.Session?) {
+        // Waiting on braze support to understand logout
+    }
+
+    func didRecieveUserNotifcation(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
```
Comment (jacobsimeon) [inline PocketKit/Sources/PocketKit/Notifications/PocketBraze.swift:87]: typo `didReceiveUserNotification`

**Review by jacobsimeon — APPROVED** [review]
Looks good, just spotted the one typo

File: PocketKit/Sources/PocketKit/Notifications/PocketBraze.swift (hunk 2 above)
Comment (bassrock) [reply PocketKit/Sources/PocketKit/Notifications/PocketBraze.swift]: ty!

**Pull request merged**